src/en/02_devenv.md

```diff
@@ -101,14 +101,29 @@ It has a fundamental role in most Rust development:
 - It’s also a front-end to run complementary tools such as those that are
   described below, in the form of sub-commands.
 
-> ### Warning
+Cargo enables automatic dependencies resolution before compilation 
+by checking their checksums.
+File `Cargo.lock` beholds all dependencies checksums which are compared to
+the one downloaded.
+When difference is detected, compilation fails:
+
+```
+error: checksum for `sha256 v1.6.0` changed between lock files
+
+this could be indicative of a few possible errors:
+
+    * the lock file is corrupt
+    * a replacement source in use (e.g., a mirror) returned a different checksum
+    * the source itself may be corrupt in one way or another
+
+unable to verify that `sha256 v1.6.0` is the same as when the lockfile was generated
+```
+
+`Cargo.lock` file is fed with dependencies' checksums when it is missing, following TOFU (*Trust On First Use*) policy.
```

Comment:
```suggestion
When the `Cargo.lock` file is missing, it is generated on first build and records the dependencies' checksums, adhering to a TOFU (*Trust On First Use*) policy.
```

Also maybe add something about generating manually with `cargo generate-lockfile`? is it really useful?